sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastNestedLoopJoinExecBase.scala

```diff
@@ -664,13 +655,23 @@ abstract class GpuBroadcastNestedLoopJoinExecBase(
             val spillableBuiltBatch = withResource(builtBatch) {
               LazySpillableColumnarBatch(_, "built_batch")
             }
-            new CrossJoinIterator(
-              spillableBuiltBatch,
-              lazyStream,
-              targetSizeBytes,
-              buildSide,
-              opTime = opTime,
-              joinTime = joinTime)
+
+            localJoinType match {
+              case LeftOuter if spillableBuiltBatch.numRows == 0 =>
```

Comment:
Do we need to worry about a full outer join?

Reply:
In the future yes, but we do not support FullOuter joins for broadcasted nested loop joins.  Support for that is s tracked by #3269.